File: app/src/main/java/com/chimu/myapp/activity/login/LoginActivity.java
package com.chimu.myapp.activity.login;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;

import com.chimu.myapp.R;
import com.chimu.mylib.base.BaseActivity;


/**
 * Created by Administrator on 2017/12/19 0019.
 */

public class LoginActivity extends BaseActivity implements LoginContract.LoginView {


    private LoginContract.LoginPresenter loginPresenter;

    @Override
    public int contentView() {
        return R.layout.activity_login;
    }

    @Override
    public void init() {
        loginPresenter = new LoginPresenterImp(this);
    }

    @Override
    public void initView() {

    }

    @Override
    public void showProgress(String s) {

    }

    @Override
    public void hintProgress() {

    }

    @Override
    public void showToast(String s) {

    }

    @Override
    public void finishView() {

    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

}

File: kotlin/src/main/java/com/chimu/kotlin/activity/loginActivity.kt
package com.chimu.kotlin.activity

import android.content.Intent
import android.os.Bundle
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.chimu.kotlin.R

/**
 *
 * kotlin实现登录逻辑
 * Created by Longwj on 2017/7/3.
 */
class loginActivity : AppCompatActivity() {
    var userName : EditText? = null
    var userPwd : EditText? = null
    var login : TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        userName = findViewById(R.id.et_account)
        userPwd = findViewById(R.id.et_password)
        login = findViewById(R.id.login)
        login!!.setOnClickListener {
            if (userName!!.text.toString() == "123" && userPwd!!.text.toString() == "123") {
                Toast.makeText(this, "login succeed1", Toast.LENGTH_SHORT).show()
                val intent = Intent(this, MainActivity::class.java)
                startActivity(intent)
            }else{
                Toast.makeText(this, "login error", Toast.LENGTH_SHORT).show()
            }
        }
    }

}